drivers/gpu/drm/vc4/vc4_hdmi.c

```diff
@@ -2028,6 +2064,20 @@ vc4_hdmi_encoder_compute_format(const struct vc4_hdmi *vc4_hdmi,
 	const struct drm_display_info *info = &connector->display_info;
 	unsigned int format;
 
+	drm_dbg(dev, "Trying with user requested output %u\n", vc4_state->requested_output_format);
+
+	format = vc4_state->requested_output_format;
+	if (vc4_hdmi_sink_supports_format_bpc(vc4_hdmi, info, mode, format, bpc)) {
+		int ret;
+
+		ret = vc4_hdmi_encoder_compute_clock(vc4_hdmi, vc4_state,
+						     mode, bpc, format);
+		if (!ret) {
+			vc4_state->output_format = format;
+			return 0;
+		}
+	}
+
```

Comment:
This patch looks mostly ok to me, but it looks like if we fail that check, we'll fallback onto the old behaviour.

I can see two things a bit wrong with that:

- I'd expect that if the user explicitly sets a format, they'd also expect to fail if we can't match that
- Also, let's assume that the user sets RGB, it doesn't work for that resolution, we fallback to YUV422. The property will still report the output format as RGB, which seems wrong?

I guess we could use a default (ie, 0) value of "auto" that would trigger the old behaviour. If the user forces it to something else, we just try that configuration and fail if we can't. What do you think?

Reply:
Sure can go for an "auto", with any other setting being explicit. That is closer to the behaviour of "Broadcast RGB" for range, so makes sense.
It may fail if hardware or the monitor is not be able to do it, but that's the user's lookout.

Reply:
If it's ever a concern, we could add another property listing the available formats based on the monitor and controller capabilities.

Reply:
I think the only case it can really fail is if trying to force RGB when 422 is required due to bandwidth limits.
The monitor should tell you what is available via the EDID, but it is overriding this when the EDID is dodgy that is the whole point of this property.

I'll try to respin in the next couple of days - rather a lot on at present.

Reply:
It's not clear what happened to this idea.